src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -53,12 +58,21 @@ public void Add(string hintName, SourceText source)
                     && c != '['
                     && c != ']'
                     && c != '{'
-                    && c != '}')
+                    && c != '}'
+                    && c != '/'
+                    && c != '\\')
                 {
                     throw new ArgumentException(string.Format(CodeAnalysisResources.HintNameInvalidChar, hintName, c, i), nameof(hintName));
                 }
             }
 
+            hintName = hintName.Replace('\\', '/');
```

Comment:
Wouldn't backslashes be allowed as part of a file or directory name on linux?

Reply:
I guess. I like that the hintName behaves the same across systems, so `dir\file.cs` will work equally everywhere, similarly as in MSBuild project files, for example.

Reply:
Thanks, I find the parallel with MSBuild convincing. Let's make sure this is documented somewhere.

Reply:
Added to documentation comments.